build_runner/lib/src/entrypoint/options.dart

```diff
@@ -521,6 +557,22 @@ class _TestCommand extends _BuildRunnerCommand {
       'Performs a single build on the specified targets and then runs tests '
       'using the compiled assets.';
 
+  @override
+  _SharedOptions _readOptions({Iterable<String> defaultBuildDirs}) {
+    // This command doesn't allow specifying directories to build, instead it
+    // always builds the `test` directory.
+    if (argResults.rest.contains('--')) {
```

Comment:
Why does including a directory to build mean that `'--'` shows up in `argResults.rest`?